NBC potential uninitialized value in alltoallv

I got this compiler warning on master this morning:

```
nbc_ialltoallv.c: In function ‘nbc_alltoallv_init’:
nbc_ialltoallv.c:274:53: warning: ‘sndext’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       char *sbuf = ((char *) sendbuf) + (sdispls[i] * sndext);
                                         ~~~~~~~~~~~~^~~~~~~~~
nbc_ialltoallv.c:50:12: note: ‘sndext’ was declared here
   MPI_Aint sndext, rcvext;
            ^~~~~~
```

This appears to be legit -- it _seems_ to stem from a distinction between MPI_IN_PLACE or not in the upper-level call to `nbc_alltoallv_init()`, but it's not immediately clear to me what the correct fix is.

@bosilca Could you have a look?

--- a/ompi/mca/coll/libnbc/nbc_ialltoallv.c
+++ b/ompi/mca/coll/libnbc/nbc_ialltoallv.c
@@ -11,6 +11,7 @@
  *                         reserved.
  * Copyright (c) 2017      IBM Corporation.  All rights reserved.
  * Copyright (c) 2018      FUJITSU LIMITED.  All rights reserved.
+ * Copyright (c) 2021      Cisco Systems, Inc.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -90,6 +91,7 @@ static int nbc_alltoallv_init(const void* sendbuf, const int *sendcounts, const 
 
     sendcounts = recvcounts;
     sdispls = rdispls;
+    sndext = rcvext;
   } else {
     res = ompi_datatype_type_extent (sendtype, &sndext);
     if (MPI_SUCCESS != res) {

